cloud: cross region replication (#8770)
pingcap/docs: tidb-cloud/changefeed-replication.md

---
title: TiDB Cloud Replication
summary: Learn how to create a replica to stream data from a primary TiDB cluster to a secondary TiDB cluster.
---

# TiDB Cloud Replication

TiDB Cloud Replication is a feature that allows you to create a continuously replicated readable secondary TiDB cluster for a primary TiDB cluster in TiDB Cloud. This readable secondary cluster (also known as cross-region replication, secondary replication, or geo-replication) can be in the same region as the primary TiDB cluster, or more commonly, in a different region.

With TiDB Cloud Replication, you can perform quick disaster recovery of a database in the event of a regional disaster or large-scale failure, which helps achieve business continuity. Once a secondary cluster is set up, you can manually initiate geographic failover to the secondary cluster in a different region.

> **Warning:**
>
> Currently, the **TiDB Cloud Replication** feature is in **Public Preview** with the following limitations:
>
> * One primary cluster can only have one replication.
> * You cannot use a secondary cluster as a source of **TiDB Cloud Replication** to another cluster.

> > * **TiDB Cloud Replication** contradicts **Sink to Apache Kafka**](/tidb-cloud/changefeed-sink-to-apache-kafka.md) and [**Sink to MySQL**](/tidb-cloud/changefeed-sink-to-mysql.md). When **TiDB Cloud Replication** is enabled, neither the primary nor the secondary cluster can use the **Sink to Apache Kafka** or **Sink to MySQL** changefeed and vice versa.

Reviewer: ```suggestion
> * **TiDB Cloud Replication** contradicts [**Sink to Apache Kafka**](/tidb-cloud/changefeed-sink-to-apache-kafka.md) and [**Sink to MySQL**](/tidb-cloud/changefeed-sink-to-mysql.md). When **TiDB Cloud Replication** is enabled, neither the primary nor the secondary cluster can use the **Sink to Apache Kafka** or **Sink to MySQL** changefeed and vice versa.
```

> > * Because TiDB Cloud uses TiCDC to establish replication, it has the same [restrictions as TiCDC](https://docs.pingcap.com/tidb/stable/ticdc-overview#restrictions).

Reviewer: Can we use internal links now that the source files are in the same repo?
```suggestion
> * Because TiDB Cloud uses TiCDC to establish replication, it has the same [restrictions as TiCDC](/ticdc/ticdc-overview.md#restrictions).
```

Author: No, we cannot use it, because "/ticdc/ticdc-overview.md#restrictions" will be replaced with "https://docs.pingcap.com/tidb-cloud/ticdc-overview.md#restrictions" in the doc building, which will cause 404.

**Planned Detach** detaches the secondary cluster from the primary cluster into an individual cluster. When **Planned Detach** is triggered, it performs the following steps:

1. Sets the primary cluster as read-only, to prevent any new transaction from being committed to the primary cluster.
2. Waits until the secondary cluster is fully synced with the primary cluster.
3. Stops the replication from the primary to the secondary cluster.
4. Sets the original secondary cluster as writable, which makes it available to serve your business.

After **Planned Detach** is finished, the original primary cluster is set as read-only. If you still need to write to the original primary cluster, you can do one of the following to set the cluster as writable explicitly:

- Go to the cluster details page, click **Settings**, and then click the **Make Writable** drop-down button.
- Connect to the SQL port of the original primary cluster and execute the following statement:

> {{< copyable "sql" >}}
> 
> ```sql
> set global tidb_super_read_only=OFF;
> ```

Reviewer: ```suggestion
    {{< copyable "sql" >}}

    ```sql
    set global tidb_super_read_only=OFF;
    ```
```

3. If the original primary cluster is still accessible, or when the original primary cluster recovers, TiDB Cloud sets it as read-only to avoid any new transaction being committed to it.

Once the original primary cluster is recovered from the outage, you still have the opportunity to review transactions that have been executed in the original primary cluster but not in the original secondary cluster by comparing the data in the two clusters, and decide whether to manually replicate these unsynchronized transactions to the original secondary cluster based on your business situation.

The data replication topology between primary and secondary clusters does not exist anymore after you detach the secondary cluster. The original primary cluster is set to the read-only mode and the original secondary cluster becomes writable. If any DML or DDL is planned on the original primary cluster, you need to disable the read-only mode manually on it by doing one of the following:

- Go to the cluster details page, click **Settings**, and then click the **Make Writable** drop-down button.
- Connect to the SQL port of the original primary cluster and execute the following statement:

> {{< copyable "sql" >}}
> 
> ```sql
> set global tidb_super_read_only=OFF;
> ```

Reviewer: ```suggestion
    {{< copyable "sql" >}}

    ```sql
    set global tidb_super_read_only=OFF;
    ```
```
